hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ReplicationManager.java

```diff
@@ -302,6 +353,35 @@ protected ContainerHealthResult processContainer(ContainerInfo containerInfo,
     return health;
   }
 
+  /**
+   * Creates a priority queue of UnderReplicatedHealthResult, where the elements
+   * are ordered by the weighted redundancy of the container. This means that
+   * containers with the least remaining redundancy are at the front of the
+   * queue, and will be processed first.
+   * @return An empty instance of a PriorityQueue.
+   */
+  protected PriorityQueue<ContainerHealthResult.UnderReplicatedHealthResult>
+      createUnderReplicatedQueue() {
+    return new PriorityQueue<>(
+        (o1, o2) -> {
+          if (o1.getWeightedRedundancy() < o2.getWeightedRedundancy()) {
+            return -1;
+          } else if (o1.getWeightedRedundancy() > o2.getWeightedRedundancy()) {
+            return 1;
+          } else {
+            // Equal weighted redundancy. Put the one with the lesser retries
+            // first
+            if (o1.getRequeueCount() < o2.getRequeueCount()) {
+              return -1;
+            } else if (o1.getRequeueCount() > o2.getRequeueCount()) {
+              return 1;
+            } else {
+              return 0;
+            }
+          }
+        });
```

Comment:
We can simplify this as

```
Comparator.comparing(UnderReplicatedHealthResult::getWeightedRedundancy)
    .thenComparing(UnderReplicatedHealthResult::getRemainingRedundancy);
```

Reply:
Oh, nice. I was not aware of this `Comparator.comparing` method. I will change this.

Reply:
Thanks.  Sorry I wrote `getRemainingRedundancy` instead of `getRequeueCount`.